### ISSUE TITLE
video freezes

### ISSUE THREAD

**Issue by bderembl** [BelledonneCommunications/linphone-desktop#46]
The video freezes during a call.
It also happens when using the 'video preview' function (in the video menu). it starts well and then it freezes. Here are the logs I get when it stops working:
```
2017-07-14 13:50:38:568 WARNING VIDIOC_QBUF 1 failed: No such device
2017-07-14 13:50:43:569 WARNING ms2017-07-14 13:50:38:568 WARNING VIDIOC_QBUF 1 failed: No such device
2017-07-14 13:50:43:569 WARNING msv4l2: still [3] buffers not dequeued at exit !
2017-07-14 13:50:43:569 ERROR VIDIOC_STREAMOFF failed: No such device
2017-07-14 13:50:43:605 MESSAGE msv4l2_thread exited.
2017-07-14 13:50:43:605 MESSAGE msv4l2 thread has joined.
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSV4L2Capture:0x56d6c30,0-->MSPixConv:0x56d6b30,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSPixConv:0x56d6b30,0-->MSTee:0x5665d90,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSTee:0x5665d90,1-->MSOGL:0x567e4a0,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSTee:0x5665d90,2-->MSJpegWriter:0x565ae40,0
2017-07-14 13:50:43:607 MESSAGE Video MSTicker thread exiting
2017-07-14 13:50:43:607 WARNING ms_yuv_buf_allocator_free(): leaving 1 mblk_t still ref'd, possible leak.
2017-07-14 13:50:43:609 MESSAGE QT: "Delete context info:" 0x5637c20
v4l2: still [3] buffers not dequeued at exit !
2017-07-14 13:50:43:569 ERROR VIDIOC_STREAMOFF failed: No such device
2017-07-14 13:50:43:605 MESSAGE msv4l2_thread exited.
2017-07-14 13:50:43:605 MESSAGE msv4l2 thread has joined.
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSV4L2Capture:0x56d6c30,0-->MSPixConv:0x56d6b30,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSPixConv:0x56d6b30,0-->MSTee:0x5665d90,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSTee:0x5665d90,1-->MSOGL:0x567e4a0,0
2017-07-14 13:50:43:605 MESSAGE ms_filter_unlink: MSTee:0x5665d90,2-->MSJpegWriter:0x565ae40,0
2017-07-14 13:50:43:607 MESSAGE Video MSTicker thread exiting
2017-07-14 13:50:43:607 WARNING ms_yuv_buf_allocator_free(): leaving 1 mblk_t still ref'd, possible leak.
2017-07-14 13:50:43:609 MESSAGE QT: "Delete context info:" 0x5637c20
```
if I close the window and try video preview again, I see a black screen with the log:
```
2017-07-14 13:52:35:790 MESSAGE Setting video size 640x480 on stream [0x561da60]
2017-07-14 13:52:35:790 ERROR Could not open /dev/video0: No such file or directory
2017-07-14 13:52:35:790 ERROR Could not open /dev/video0: No such file or directory
2017-07-14 13:52:35:790 MESSAGE ogl_display_new : 0x318b3e0

2017-07-14 13:52:35:790 ERROR no such method on filter MSOGL, fid=16387 method index=1
2017-07-14 13:52:35:790 MESSAGE ms_filter_link: MSV4L2Capture:0x561bc40,0-->MSPixConv:0x5634f60,0
2017-07-14 13:52:35:790 MESSAGE ms_filter_link: MSPixConv:0x5634f60,0-->MSTee:0x318b590,0
2017-07-14 13:52:35:790 MESSAGE ms_filter_link: MSTee:0x318b590,1-->MSOGL:0x561aed0,0
2017-07-14 13:52:35:790 MESSAGE ms_filter_link: MSTee:0x318b590,2-->MSJpegWriter:0x55f8d60,0
2017-07-14 13:52:35:790 MESSAGE Video MSTicker priority left to normal.
2017-07-14 13:52:35:790 MESSAGE msv4l2_thread starting
2017-07-14 13:52:35:791 ERROR Could not open /dev/video0: No such file or directory
2017-07-14 13:52:35:791 WARNING msv4l2 could not be openned
2017-07-14 13:52:35:791 MESSAGE msv4l2_thread exited.
2017-07-14 13:52:35:792 MESSAGE QT: Thread QThread(0x1d7a7d0) "Set context info (width: 590, height: 395):" 0x562a330
2017-07-14 13:52:35:792 MESSAGE set native window id: 0x562a330 (width: 590, height: 395)

2017-07-14 13:52:35:792 ERROR no such method on filter MSV4L2Capture, fid=16387 method index=4
2017-07-14 13:52:35:792 MESSAGE init opengles_display (590 x 395, gl initialized:0)
2017-07-14 13:52:35:793 MESSAGE resize opengles_display (590 x 395, gl initialized:0)
2017-07-14 13:52:35:803 MESSAGE OpenGL program info: [NO INFORMATION]
2017-07-14 13:52:38:490 MESSAGE msv4l2 thread has joined.
2017-07-14 13:52:38:490 MESSAGE ms_filter_unlink: MSV4L2Capture:0x561bc40,0-->MSPixConv:0x5634f60,0
2017-07-14 13:52:38:490 MESSAGE ms_filter_unlink: MSPixConv:0x5634f60,0-->MSTee:0x318b590,0
2017-07-14 13:52:38:490 MESSAGE ms_filter_unlink: MSTee:0x318b590,1-->MSOGL:0x561aed0,0
2017-07-14 13:52:38:490 MESSAGE ms_filter_unlink: MSTee:0x318b590,2-->MSJpegWriter:0x55f8d60,0
2017-07-14 13:52:38:490 MESSAGE Video MSTicker thread exiting
2017-07-14 13:52:38:493 MESSAGE QT: "Delete context info:" 0x562a330
```

I am using gnome 3.24.2

**Comment by bderembl** [comment]
Hi,
I was wondering if you had any chance to look at this issue: this is really problematic as it makes linphone unusable for me.
thanks

**Comment by Wescoeur** [comment]
It seems your webcam cannot be opened by mediastreamer2.
Is it activated and not used by other program?

**Comment by bderembl** [comment]
I am not sure this is the right explanation: the video actually works for the first 30 seconds of the call.
it isn't used by any other program

**Comment by bderembl** [comment]
it seems to work now.
I wonder if this commit fixed it 8a84a26cfaf6bcd05fc12f127a8c63246ad89f0f
